BootloaderCommonPkg/Include/Guid/SmmInformationGuid.h

```diff
@@ -16,16 +16,52 @@
 #define __SMM_INFORMATION_GUID_H__
 
 ///
-/// Serial Port Information GUID
+/// SMM Information GUID
 ///
 extern EFI_GUID gSmmInformationGuid;
 
+//
+// Set this flag if 4KB SMRAM at SmmBase used for communication
+// between bootloader and UEFI payload.
+//
+#define   SMM_FLAGS_4KB_COMMUNICATION  BIT0
+
+typedef enum {
+  MEM,
+  IO,
+  PCICFG
+} REG_TYPE;
+
+typedef enum {
+  WIDE8,
+  WIDE16,
+  WIDE32
+} REG_WIDTH;
+
+#pragma pack(1)
+
+///
+/// Generic Address Space
+///
+typedef struct {
+  UINT8   RegType;
+  UINT8   RegWidth;
+  UINT8   SmiGblPos;
+  UINT8   SmiApmPos;
+  UINT8   SmiEosPos;
+  UINT8   Rsvd[3];
+  UINT32  Address;
+} SMI_CTRL_REG;
```

Comment:
Where these fields are filled in this patch?

Reply:
Will add changes to APL platform package in the open source to show how these fields are used.